### ISSUE TITLE
[suggestion] make encoder/decoder configurable to avoid conflicts with eiconv

### ISSUE THREAD

**Issue by fcevado** [gen-smtp/gen_smtp#334]
When using a different encoding/decoding library than `eiconv` i need to make a fake module that implements the `convert/3`. if a dependency uses eiconv, it might lead co conflicts.

my suggestion is to have it being configurable, so `gen_smtp` uses the configured module, otherwise keep using eiconv.
for the sake of documenting it would be good to have a behaviour defining a contract for `convert/3`.

**Comment by Vagabond** [comment]
That should be relatively easy to do. You could just add a new option key to mimemail's options tuple and thread it through to where it calls mimemail:convert/3

**Comment by fcevado** [comment]
yep, i'm planning to work on this this weekend, just asking if the approach is approved. should i include the behavior? probably `mimemail_encoder` or something like that
